fast/stages/3-network-security/main.tf

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+# tfdoc:file:description Next-Generation Firewall Enterprise configuration.
+
+locals {
+  enabled_vpcs = {
+    dev-spoke-0  = try(var.vpc_self_links.dev-spoke-0, null) != null
+    prod-spoke-0 = try(var.vpc_self_links.prod-spoke-0, null) != null
+  }
+  vpc_ids = {
+    for k, v in var.vpc_self_links
+    : k => replace(v, "https://www.googleapis.com/compute/v1/", "")
+  }
+}
+
+# Dedicated quota project for ngfw enterprise endpoints
+module "ngfw-quota-project" {
+  count           = var.ngfw_enterprise_config.quota_project_id == null ? 1 : 0
+  source          = "../../../modules/project"
+  billing_account = var.billing_account.id
+  name            = "net-ngfw-0"
+  parent          = var.folder_ids.networking
+  prefix          = var.prefix
+  services        = ["networksecurity.googleapis.com"]
+}
+
+module "ext-ngfw-quota-project" {
```

Comment:
this should use the same module as the one used for creation, leveraging `project_create` and nulling billing and parent fields, etc.

Reply:
This is different from the one above. It is only used to enable the networksecurity api in an already existing project, created outside of here. Notice here project_create is false. I agree we should null other fields. I'll do that.

Reply:
My point exactly. Use a single module invocation, with project_create false or true, and billing/parent null if create is not needed. The module takes care of the rest.